src/sentry/search/utils.py

```diff
@@ -117,32 +117,41 @@ def parse_datetime_comparison(value):
 
 def parse_datetime_value(value):
     # timezones are not supported and are assumed UTC
-    if value[-1] == 'Z':
+    if value[-1:] == 'Z':
         value = value[:-1]
 
+    result = None
+
     # A value that only specifies the date (without a time component) should be
     # expanded to an interval that spans the entire day.
-    if len(value) in (8, 10):
```

Comment:
This conditional caused timestamps between 2001/09/09 and 2286/11/20 to not be usable unless you either zero-prefixed them or added a fractional component to avoid these length checks.